sdk/src/wallet/account/operations/transaction/high_level/minting/create_native_token.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use primitive_types::U256;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    client::{
+        api::{PreparedTransactionData, PreparedTransactionDataDto},
+        secret::SecretManage,
+    },
+    types::block::{
+        address::AliasAddress,
+        output::{
+            feature::MetadataFeature, unlock_condition::ImmutableAliasAddressUnlockCondition, AliasId,
+            AliasOutputBuilder, FoundryId, FoundryOutputBuilder, Output, SimpleTokenScheme, TokenId, TokenScheme,
+        },
+    },
+    wallet::account::{
+        types::{Transaction, TransactionDto},
+        Account, TransactionOptions,
+    },
+};
+
+/// Address and foundry data for `create_native_token()`
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateNativeTokenParams {
+    /// The alias id which should be used to create the foundry.
+    pub alias_id: Option<AliasId>,
+    /// Circulating supply
+    pub circulating_supply: U256,
+    /// Maximum supply
+    pub maximum_supply: U256,
+    /// Foundry metadata
+    #[serde(with = "crate::utils::serde::option_prefix_hex_vec")]
+    pub foundry_metadata: Option<Vec<u8>>,
+}
+
+/// The result of a minting native token transaction
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MintNativeTokenTransaction {
```

Comment:
Shouldn't this also be renamed now?
```suggestion
pub struct CreateNativeTokenTransaction {
```

Reply:
nope, because this type is for creating **and** for minting and since creation always contains a minting I kept this name.